ynput/ayon-nuke
# AY-971 Refactor staging directory handling in Nuke creators

# Change description
resolve https://github.com/ynput/ayon-nuke/issues/21

Implementing custom staging directory for all of Nuke products.

Bugfixes:
* `scriptmenus` still loading `openpype` modules
* selection validation code for all creators
* update/remove on instance from manually deleted nodes
* temporary render path settings ignored 

# Additional informations
- implementing changes from https://github.com/ynput/ayon-core/pull/887


# Testing notes

1. Enable custom staging directory with some profiles `ayon+settings://core/tools/publish/custom_staging_dir_profiles` (make it persistent so you can troubleshoot)
2. Create a template matching the staging directory under `ayon+anatomy://Dummy/templates/staging`
e.g. `{root[work]}/{project[name]}/{hierarchy}/{folder[name]}/temp_render_only/{product[type]}/{product[name]}`

3. Open Nuke and create+publish some products
4. Ensure that the staging directory is working is properly redirected to the relevant template for the associated products/task
